.github/workflows/freertos_demos.yml

```diff
@@ -66,10 +68,44 @@ jobs:
           sudo apt-get -y update
           sudo apt-get -y install build-essential
 
-      - name: Build Posix_GCC Demo
+      - name: Build Posix_GCC Full Demo
+        id: build-posix-gcc-demo
         shell: bash
         working-directory: FreeRTOS/Demo/Posix_GCC
-        run: make -j
+        run: |
+          make -j
+
+      - name: Run and monitor Posix_GCC Full Demo
+        if: success()
+        working-directory: FreeRTOS/Demo/Posix_GCC
+        run: |
+          timeout 5m build/posix_demo | tee output.txt
+          if grep -q "OK: No errors" output.txt; then
+            echo -e "\nSUCCESS - Found SUCCESS string in output."
+          else
+            echo -e "\nFAILURE - SUCCESS string not found in output."
+            exit 1
+          fi
+
+      - name: Build Posix_GCC Blinky Demo
+        if: success() || failure() && steps.build-posix-gcc-demo.outcome == 'success'
+        shell: bash
+        working-directory: FreeRTOS/Demo/Posix_GCC
+        run: |
+          sed -i "s/#define[[:space:]]\+mainSELECTED_APPLICATION[[:space:]]\+FULL_DEMO/#define    mainSELECTED_APPLICATION      BLINKY_DEMO/g" main.c
+          make -j
+
+      - name: Run and monitor Posix_GCC Blinky Demo
+        if: success() || failure() && steps.build-posix-gcc-demo.outcome == 'success'
```

Comment:
```suggestion
        if: success() || failure() && steps.build-posix-gcc-blinky-demo.outcome == 'success'
```

Reply:
This will make it so this only runs as long as the build for the blinky demo succeeds if you commit my previous suggestion.